src/server/config/config.js

```diff
@@ -163,8 +187,23 @@ module.exports = class Config {
 
   getSchema() {
     if (!this[schema]) {
-      let objKeys = zipObject([...this[schemaKeys]]);
-      this[schema] = Joi.object().keys(objKeys).default();
+      this[schema] = (function convertToSchema(children) {
+        let objKeys = zipObject([...children]);
```

Comment:
One of the tests checks that Config throws when not given a schema but a value is set, which seems correct as it's delegating to Joi, and Joi should be making sure you don't set a value that's not part of the schema. 

You're right that this code is wrong, but setting the `.keys` on the Joi objects is still required to get that validation.